File: Week 6/ahmtfurknun-cs50-problems-2021-x-sentimental-cash/cash.py

while True:
    n = input("Change owed: ")
    try:
        n = float(n)
        if not n > 0:
            continue
        else:
            n = float(n) * 100
            break
    except:
        continue
    
quarters = dimes = nickels = pennies = 0

while n != 0:
    if n >= 25:
        quarters += 1
        n -= 25
        
    elif n >= 10:
        dimes += 1
        n -= 10
        
    elif n >= 5:
        nickels += 1
        n -= 5
        
    else:
        pennies += 1
        n -= 1


print(quarters+dimes+nickels+pennies)        

File: Week 1/ahmtfurknun-cs50-problems-2020-x-cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);

    int cents = round(dollars * 100);

    int quarters = 0;
    int dimes = 0;
    int nickels = 0;
    int pennies = 0;

    while (cents >= 25)
    {
        quarters = quarters + 1;
        cents = cents - 25;
    }

    while (cents >= 10)
    {
        dimes = dimes + 1;
        cents = cents - 10;
    }

    while (cents >= 5)
    {
        nickels = nickels + 1;
        cents = cents - 5;
    }

    while (cents >= 1)
    {
        pennies = pennies + 1;
        cents = cents - 1;
    }


    int tot = quarters + dimes + nickels + pennies;
    printf("%i\n", tot);
}